sso-plugin/optimize-sso-keycloak/README.md

```diff
@@ -0,0 +1,80 @@
+# Camunda Optimize Keycloak SSO Example
+
+This example demonstrates how you can enable SSO with Optimize and Keycloak.
+
+The most important part from the Optimize perspective of this example is the custom Optimize SSO Plugin in:
+[./optimize/sso-keycloak-example-plugin/](./optimize/sso-keycloak-example-plugin/)
+To find out more about the Optimize SSO Plugin mechanism, find the docs [here](https://docs.camunda.org/optimize/latest/technical-guide/import/plugins/#single-sign-on)
+
+It includes a docker-compose with:
+1. Camunda Optimize
+2. ElasticSearch
+3. Camunda BPM
+2. Keycloak Authentication Server
+3. Keycloak Proxy Server
+
+The Keycloak Server has one user:
+
+```
+demo:notdemo
+```
+
+# How to run?
+
+## Clone the Repo
+
+## Download and Configure Optimize
+
+1. Download https://camunda.org/enterprise-release/optimize/2.3.0/camunda-optimize-2.3.0-standalone.zip to folder ./optimize/
+2. Rename to `camunda-optimize.zip`
+3. Place Optimize license in OptimizeLicense.txt
+3. Run `mvn package` in `./optimize/sso-keycloak-example-plugin`
+
+## Run docker-compose
+
+1. Build all images with `docker-compose build`
+1. Login to private Camunda Docker EE Registry with `docker login registry.camunda.cloud` Use your EE LDAP credentials to log in.
+1. Start all images `docker-compose up -d`
+
+## Open Optimize
+
+1. Open WebBrowser
+2. Open `localhost:8095`
+3. Login with: `demo:notdemo`
+
+# Show me some some code
+
+```java
+package com.camunda.optimize.plugin.sso;
+
+import java.util.Enumeration;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.camunda.optimize.plugin.security.authentication.AuthenticationExtractor;
+import org.camunda.optimize.plugin.security.authentication.AuthenticationResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeycloakAuthenticationProvider implements AuthenticationExtractor {
+
+	private Logger logger = LoggerFactory.getLogger(getClass());
+
+	public AuthenticationResult extractAuthenticatedUser(HttpServletRequest request) {
+		AuthenticationResult result = new AuthenticationResult();
+		String user = request.getHeader("KEYCLOAK_USERNAME");
```

Comment:
maybe a short explanation section before the plugin code on what the overall setup actually is and what keycloak does and then what the optimize plugin does may help the users to understand the scenario better?

Reply:
added an explanation